### DISCUSSION TITLE
Force textbox single line

### DISCUSSION THREAD

**Discussion by tgbv** [fabricjs/fabric.js#6787] [Q&A]
Hi! How can I force a textbox to be single line on editing?

**Comment by asturur** [comment]
Can you be more specific?
I would think you would use the IText instead of the TextBox?

**Comment by tgbv** [comment]
When I use Textbox on canvas, and user edits it, I need to prevent him from
typing new lines.

I believe that issue occurs on IText as well.

**Comment by asturur** [comment]
How would you do that in a normal textarea in the DOM? if you can set up
that, we can probably transfer it to the textbox hidden textarea

**Comment by tgbv** [comment]
As far as my research goes, there's no way to prevent textarea from going single-line mode.

I've tried it with regex to strip the newline characters but it works only partially (after second line):
```
Canvas.on('text:editing:entered', e=>{
	let t = e.target

        t.fill = 'black'
        t.text = ''
        t.hiddenTextarea.value = ''
        t.hiddenTextarea.setAttribute('maxlength', 25)

        // here
        t.hiddenTextarea.setAttribute('oninput', "this.value = this.value.replace(/\\n/g,'')")
})
```

**Comment by asturur** [comment]
if you can write an event handler for javascript to block user inputs of newlines, then we can add it to the fabric textbox somehow. Otherwise someone has to find a way to do that first, but i won't have time for it for sure.
